File: report/src/main/java/com/vmlens/report/runelementtype/ReportLockType.java
package com.vmlens.report.runelementtype;

public enum ReportLockType {

    MONITOR("Monitor","Monitor") ,
    REENTRANT_LOCK("ReentrantLock", "WriteLock") ,
    READ_LOCK("ReadLock" , "AtomicRead") ,
    WRITE_LOCK("WriteLock" , "WriteLock");

    private final String text;
    private final String textForMethodWithLock;

    ReportLockType(String text, String textForMethodWithLock) {
        this.text = text;
        this.textForMethodWithLock = textForMethodWithLock;
    }

    public String text() {
        return text;
    }

    public String textForMethodWithLock() {
        return textForMethodWithLock;
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/ReportLockType.scala
package com.anarsoft.race.detection.report.element.runelementtype

enum ReportLockType(val text: String, val textForMethodWithLock: String) {
  case MONITOR extends ReportLockType("Monitor", "Monitor")
  case REENTRANT_LOCK extends ReportLockType("ReentrantLock", "WriteLock")
  case READ_LOCK extends ReportLockType("ReadLock", "AtomicRead")
  case WRITE_LOCK extends ReportLockType("WriteLock", "WriteLock")
}